src/Features/Core/Portable/Diagnostics/DiagnosticAnalyzerService_IncrementalAnalyzer.cs

```diff
@@ -218,7 +197,8 @@ internal BaseDiagnosticIncrementalAnalyzer Analyzer
 
             private BaseDiagnosticIncrementalAnalyzer GetAnalyzer(bool useV2)
             {
-                return useV2 ? (BaseDiagnosticIncrementalAnalyzer)_engineV2 : _engineV1;
+                // v1 engine has been removed, always use v2 engine.
```

Comment:
if you leave the code I asked you to leave, then, it can't return same analyzer otherwise, when option is changed, it will try to turn off that analyzer which is same analyzer as the other one.